src/core/server/context/container/context.ts

```diff
@@ -224,36 +226,51 @@ export class ContextContainer implements IContextContainer {
     }
 
     return (async (...args: HandlerParameters<RequestHandler>) => {
-      const context = await this.buildContext(source, ...args);
+      const context = this.buildContext(source, ...args);
       return handler(context, ...args);
     }) as (
       ...args: HandlerParameters<RequestHandler>
     ) => ShallowPromise<ReturnType<RequestHandler>>;
   };
 
-  private async buildContext(
+  private buildContext(
     source: symbol,
     ...contextArgs: HandlerParameters<RequestHandler>
-  ): Promise<HandlerContextType<RequestHandler>> {
+  ): HandlerContextType<RequestHandler> {
     const contextsToBuild = new Set(this.getContextNamesForSource(source));
+    const builtContextPromises: Record<string, Promise<unknown>> = {};
+
+    const builtContext = {} as HandlerContextType<RequestHandler>;
+    (builtContext as unknown as RequestHandlerContext).resolve = async (keys) => {
+      const resolved = await Promise.all(
```

Comment:
Do we want to reject everything on the first rejected promise and/or error?

Reply:
Yea, I think we do. Managing partially resolved contexts seems too complex and not really necessary given there's no real risk given the underlying providers. This is why I felt a simple `Promise.all` was good enough.